### DISCUSSION TITLE
Automatic avif and webp detection

### DISCUSSION THREAD

**Discussion by marcuslindblom** [imgproxy/imgproxy#898] [Q&A]
We use Bunny CDN and their vary logic only works if a known image extension is included in the url. As far as I know, imgproxy can only handle automatic detection if the extension is left out. Is that correct or is there a workaround?

**Comment by DarthSim** [comment]
Hey @marcuslindblom!

`IMGPROXY_ENABLE_WEBP_DETECTION` and `IMGPROXY_ENABLE_AVIF_DETECTION` work only when the resultant image format isn't set explicitly (i.e. extension is ommited). `IMGPROXY_ENFORCE_WEBP` and `IMGPROXY_ENFORCE_AVIF` work even if the resultant image format is set explicitly.
